plugins/inputs/nginx/nginxplus.go

```diff
@@ -0,0 +1,483 @@
+		} `json:"miss"`
+		Expired struct {
+			Responses        int64 `json:"responses"`
+			Bytes            int64 `json:"bytes"`
+			ResponsesWritten int64 `json:"responses_written"`
+			BytesWritten     int64 `json:"bytes_written"`
+		} `json:"expired"`
+		Bypass struct {
+			Responses        int64 `json:"responses"`
+			Bytes            int64 `json:"bytes"`
+			ResponsesWritten int64 `json:"responses_written"`
+			BytesWritten     int64 `json:"bytes_written"`
+		} `json:"bypass"`
+	} `json:"caches"`
+
+	Stream struct {
+		ServerZones map[string]struct {
+			Processing  int `json:"processing"`
+			Connections int `json:"connections"`
+			Sessions    *struct {
+				Total       int64 `json:"total"`
+				Sessions1xx int64 `json:"1xx"`
+				Sessions2xx int64 `json:"2xx"`
+				Sessions3xx int64 `json:"3xx"`
+				Sessions4xx int64 `json:"4xx"`
+				Sessions5xx int64 `json:"5xx"`
+			} `json:"sessions"`
+			Discarded *int64 `json:"discarded"` // added in version 7
+			Received  int64  `json:"received"`
+			Sent      int64  `json:"sent"`
+		} `json:"server_zones"`
+		Upstreams map[string]struct {
+			Peers []struct {
+				ID            int    `json:"id"`
+				Server        string `json:"server"`
+				Backup        bool   `json:"backup"`
+				Weight        int    `json:"weight"`
+				State         string `json:"state"`
+				Active        int    `json:"active"`
+				Connections   int64  `json:"connections"`
+				ConnectTime   *int   `json:"connect_time"`
+				FirstByteTime *int   `json:"first_byte_time"`
+				ResponseTime  *int   `json:"response_time"`
+				Sent          int64  `json:"sent"`
+				Received      int64  `json:"received"`
+				Fails         int64  `json:"fails"`
+				Unavail       int64  `json:"unavail"`
+				HealthChecks  struct {
+					Checks     int64 `json:"checks"`
+					Fails      int64 `json:"fails"`
+					Unhealthy  int64 `json:"unhealthy"`
+					LastPassed *bool `json:"last_passed"`
+				} `json:"health_checks"`
+				Downtime  int64 `json:"downtime"`
+				Downstart int64 `json:"downstart"`
+				Selected  int64 `json:"selected"`
+			} `json:"peers"`
+			Zombies int `json:"zombies"`
+		} `json:"upstreams"`
+	} `json:"stream"`
+}
+
+func gatherStatusUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
+	if err := dec.Decode(status); err != nil {
+		return fmt.Errorf("Error while decoding JSON response")
+	}
+	status.Gather(tags, acc)
+	return nil
+}
+
+func (s *Status) Gather(tags map[string]string, acc telegraf.Accumulator) {
+	s.gatherProcessesMetrics(tags, acc)
+	s.gatherConnectionsMetrics(tags, acc)
+	s.gatherSslMetrics(tags, acc)
+	s.gatherRequestMetrics(tags, acc)
+	s.gatherZoneMetrics(tags, acc)
+	s.gatherUpstreamMetrics(tags, acc)
+	s.gatherCacheMetrics(tags, acc)
+	s.gatherStreamMetrics(tags, acc)
+}
+
+func (s *Status) gatherProcessesMetrics(tags map[string]string, acc telegraf.Accumulator) {
+	acc.AddFields(
+		"nginx.processes",
```

Comment:
Anywhere we are using dotted notation should either use snake_case or in some cases the subitems should become tags.  I think in this PR all should become snake_case.

This applies to measurement names and field names throughout the pull request.